dist/source/wrapper/DdUrlTransfer.brs

```diff
@@ -0,0 +1,601 @@
+            if (msg <> invalid)
+                msgType = type(msg)
+                if (msgType = "roUrlEvent")
+                    if (msg.GetInt() = 1) ' transfer complete
+                        durationMs& = timer.TotalMilliseconds()
+                        transferTime# = millisToSec(durationMs&)
+                        response = msg.GetString()
+                        bytesDownloaded = Len(response)
+                        httpCode = msg.GetResponseCode()
+                        status = "ok"
+                        if (httpCode < 0)
+                            status = msg.GetFailureReason()
+                        end if
+                        resource = {
+                            url: url
+                            method: "GET"
+                            transferTime: transferTime#
+                            httpCode: httpCode
+                            status: status
+                            bytesDownloaded: bytesDownloaded
+                            traceId: traceId
+                            spanId: spanId
+                        }
+                        m.datadogRumAgent.callfunc("addResource", resource)
+                        return response
+                    else
+                        ddLogWarning("Got roUrlEvent " + FormatJson(msg))
+                    end if
+                else
+                    ddLogWarning("Got unexpected msg " + FormatJson(msg))
+                end if
+            end if
+        end while
+        return ""
+    end function
+    ' ----------------------------------------------------------------
+    ' Connect to the remote service as specified in the URL and write
+    ' the response body to a file on the Roku device's filesystem. This
+    ' function does not return until the exchange is complete and may
+    ' block for a long time. The HTTP response code from the server is
+    ' returned. It is not possible to access any of the response headers.
+    ' If this information is required use the AsyncGetToFile() method instead.
+    '
+    ' @param filename (string) The file on the Roku device's filesystem
+    ' to which the response body is to be written
+    '
+    ' @return (string) The HTTP response code
+    ' ----------------------------------------------------------------
+    instance.GetToFile = function(filename as string) as integer
+        timer = CreateObject("roTimespan")
+        port = CreateObject("roMessagePort")
+        m.roUrlTransfer.SetMessagePort(port)
+        url = m.roUrlTransfer.GetUrl()
+        traceId = generateUniqueId()
+        spanId = generateUniqueId()
+        m.roUrlTransfer.AddHeader("x-datadog-trace-id", traceId)
+        m.roUrlTransfer.AddHeader("x-datadog-parent-id", spanId)
+        m.roUrlTransfer.AddHeader("x-datadog-sampling-priority", "1")
+        m.roUrlTransfer.AddHeader("x-datadog-origin", "rum")
+        timer.Mark()
+        result = m.roUrlTransfer.AsyncGetToFile(filename)
```

Comment:
why do you need to methods for each `GET`, `POST` request type ? Why not just using the `*toString` variant ?

Reply:
Because I’m creating a wrapper around the standard network client, and so I’m creating the same function flavors as the official client uses